http-client.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+type Proxy struct {
+	URL      string
+	Username string
+	Password string
+}
+
+func (p Proxy) HasAuthorizationCredentials() bool {
+	return p.Username != "" && p.Password != ""
+}
+
+func (p Proxy) BasicAuthorizationHeader() string {
+	auth := fmt.Sprintf("%s:%s", p.Username, p.Password)
+	encodedAuth := base64.StdEncoding.EncodeToString([]byte(auth))
+	return "Basic " + encodedAuth
+}
+
+func (p Proxy) HttpTransport() http.Transport {
+	proxyURLFunc := func(r *http.Request) (*url.URL, error) {
+		return url.Parse(p.URL)
+	}
+	transport := http.Transport{
```

Comment:
Scanner specifies
InsecureSkipVerify: true
in http.Transport so sigstore-interface should specify it as well